### PR TITLE
(#5612) - silence "vm" warnings, verify build, fix build deps

### PR DESCRIPTION
**Pull request by nolanlawson**
In the course of trying to silence `vm` warnings (#5612), I realized that #5653 actually introduced a subtle problem whereby sub-packages are aggressively bundled (e.g. `pouchdb-mapreduce` will bundle `pouchdb-utils` into itself), which would be a problem for custom builds because presumably in those cases people want module-deduping more than they want rollup-inlining.

In the course of fixing that issue, I also went ahead and added a new test, `npm run verify-build` (replacing and enhancing `npm run verify-bundle-size`), which checks that the various sub-packages were built correctly. This should avoid accidentally introducing a change that breaks this in the future.

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -159,6 +159,12 @@ There's a WebSQL storage quota test available in:
 
 Run `npm run dev`, then open it in Safari or iOS.
 
+### Build tests
+
+To verify that the build was done correctly, there are some tests here:
+
+    npm run verify-build
+
```
Comment (nolanlawson) [inline TESTING.md]: verify-bundle-size was not documented

**Comment by daleharvey** [comment]
The extra test is great, sounds like this has been a hassle so good to get some assurance it wont regret, needs rebasing however but 👍

**Comment by nolanlawson** [comment]
Yeah also the added test suite makes it so that if I fix #5626 I'll be able to add a test to ensure it actually worked.

**Comment by nolanlawson** [comment]
Squash-merged as 1a589fdd5e0a183354e380a0eb84734b07972dc4

**Comment by nolanlawson** [comment]
Oops, #5612 was the wrong issue to link this to.

**Comment by nolanlawson** [comment]
Made a mistake during rebase, fixed it with 4cd2a1fa88a8b3de0f4dfe731620c89d46e1c5d8

**Pull request closed without merging**